mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -131,10 +132,14 @@ getObject' =
 
 read :: Address -> AWS (Maybe Text)
 read a = do
-  resp <- getObject' a
-  let format y = T.concat . TL.toChunks . TL.decodeUtf8 $ y
-  z <- liftIO . sequence $ (\r -> runResourceT (r ^. gorsBody . to bodyResponse $$+- sinkLbs)) <$> resp
-  pure (format <$> z)
+  r <- read' a
+  z <- liftIO . sequence $ (runResourceT . ($$+- sinkLbs)) <$> r
+  pure $ fmap (T.concat . TL.toChunks . TL.decodeUtf8) z
+
+read' :: Address -> AWS (Maybe (ResumableSource (ResourceT IO) BS.ByteString))
```

Comment:
Would have to rewrite read in the test, which would be the same as testing read which is already tested [here](https://github.com/ambiata/mismi/blob/master/mismi-s3/test/Test/IO/Mismi/S3/Commands.hs)